### PR TITLE
Localization: add other pyramid map files for pyramid_map module

### PR DESCRIPTION
**Pull request by tthhee**

### CODE REVIEW

File: modules/localization/msf/local_pyramid_map/pyramid_map/pyramid_map_node_config.cc [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+namespace apollo {
+namespace localization {
+namespace msf {
+
+PyramidMapNodeConfig::PyramidMapNodeConfig() {}
+
+PyramidMapNodeConfig::~PyramidMapNodeConfig() {}
+
+BaseMapNodeConfig* PyramidMapNodeConfig::Clone() {
+  PyramidMapNodeConfig* map_node_config = new PyramidMapNodeConfig();
```
Comment (jinghaomiao) [inline modules/localization/msf/local_pyramid_map/pyramid_map/pyramid_map_node_config.cc]: Use shared pointer.

Comment (tthhee) [reply modules/localization/msf/local_pyramid_map/pyramid_map/pyramid_map_node_config.cc]: fixed

File: modules/localization/msf/local_pyramid_map/pyramid_map/pyramid_map_node.cc [hunk 2]
```diff
@@ -0,0 +1,484 @@
...
+namespace msf {
+
+PyramidMapNode::PyramidMapNode() {}
+
+PyramidMapNode::~PyramidMapNode() {}
+
+void PyramidMapNode::Init(const BaseMapConfig* map_config) {
+  map_config_ = map_config;
+
+  map_node_config_ = new PyramidMapNodeConfig();
```
Comment (jinghaomiao) [inline modules/localization/msf/local_pyramid_map/pyramid_map/pyramid_map_node.cc]: Use safe pointer when necessary.

File: modules/localization/msf/local_pyramid_map/pyramid_map/pyramid_map_node.cc [hunk 3]
```diff
@@ -0,0 +1,484 @@
...
+  } else {
+    map_node_config_->has_map_version_ = true;
+    map_node_config_->has_body_md5_ = true;
+  }
+
+  is_reserved_ = false;
+  data_is_ready_ = false;
+  is_changed_ = false;
+
+  map_matrix_ = new PyramidMapMatrix();
```
Comment (jinghaomiao) [inline modules/localization/msf/local_pyramid_map/pyramid_map/pyramid_map_node.cc]: Use safe pointer

File: modules/localization/msf/local_pyramid_map/pyramid_map/pyramid_map_node.cc [hunk 4]
```diff
@@ -0,0 +1,484 @@
...
+  map_matrix_handler_ =
+      PyramidMapMatrixHandlerSelector::AllocPyramidMapMatrixHandler(
+          map_node_config_->map_version_);
+  compression_strategy_ = new ZlibStrategy();
```
Comment (jinghaomiao) [inline modules/localization/msf/local_pyramid_map/pyramid_map/pyramid_map_node.cc]: Use safe pointer

File: modules/localization/msf/local_pyramid_map/pyramid_map/pyramid_map_node.cc [hunk 5]
```diff
@@ -0,0 +1,484 @@
...
+  }
+
+  InitMapMatrix(map_config_);
+}
+
+void PyramidMapNode::Init(const BaseMapConfig* map_config,
+                          const MapNodeIndex& index, bool create_map_cells) {
+  map_config_ = map_config;
+
+  map_node_config_ = new PyramidMapNodeConfig();
```
Comment (jinghaomiao) [inline modules/localization/msf/local_pyramid_map/pyramid_map/pyramid_map_node.cc]: Use safe pointer. It applies to all "new" pointers.

Comment (tthhee) [reply modules/localization/msf/local_pyramid_map/pyramid_map/pyramid_map_node.cc]: fixed

File: modules/localization/msf/local_pyramid_map/pyramid_map/pyramid_map_node.cc (hunk 4 above)
Comment (tthhee) [reply modules/localization/msf/local_pyramid_map/pyramid_map/pyramid_map_node.cc]: fixed

File: modules/localization/msf/local_pyramid_map/pyramid_map/pyramid_map_node.cc (hunk 3 above)
Comment (tthhee) [reply modules/localization/msf/local_pyramid_map/pyramid_map/pyramid_map_node.cc]: fixed

File: modules/localization/msf/local_pyramid_map/pyramid_map/pyramid_map_node.cc (hunk 2 above)
Comment (tthhee) [reply modules/localization/msf/local_pyramid_map/pyramid_map/pyramid_map_node.cc]: fixed

**Comment by TommiTallgren** [comment]
>Softagram Impact report
This was done to help the discussion if automated analysis (this done without build information) could help the project (free for open-source).

**Pull request merged**